tests/route/test_route_perf.py

```diff
@@ -276,19 +315,19 @@ def test_perf_add_remove_routes(
     ptfadapter,
     enum_rand_one_per_hwsku_frontend_hostname,
     request,
+    check_config,
     ip_versions,
     enum_rand_one_frontend_asic_index,
 ):
     duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
     asichost = duthost.asic_instance(enum_rand_one_frontend_asic_index)
+    max_scale = request.config.getoption("--max_scale")
     # Number of routes for test
     set_num_routes = request.config.getoption("--num_routes")
-    if set_num_routes is None:
-        topo_name = tbinfo["topo"]["name"]
-        if topo_name in ["m0", "mx"]:
-            set_num_routes = DEAFULT_M0_MX_NUM_ROUTES
-        else:
-            set_num_routes = DEFAULT_NUM_ROUTES
+    if max_scale and set_num_routes is not None:
+        raise Exception("--max_scale and --num_routes are mutually exclusive")
```

Comment:
@prabhataravind could we have max_scale by default, and num_routes as on demand, if there is a conflict, should we honor num_routes since which is on demand?

Reply:
@StormLiangMS, we didn't use max_scale by default as the test might take a long time to finish for certain platforms/skus where the CRM available counters are large. This may not be ideal for nightly runs. Our idea was that "max_scale" could be used on an "as needed" basis.

Reply:
Also, the CRM available counters don't seem to be fully reliable and I have a case open on this with bcm. Their take is that it is a best effort estimate. For nightly tests, it might be better to have a deterministic number to scale up to.

Reply:
@prabhataravind I see, so both num_routes and max_scale are not default, but are on demand? Which make them mutually exclusive?

Reply:
@StormLiangMS, Yes, that's correct. Both these are optional arguments to the script. The default value will be taken from the lookup table.